pylint/test/input/func_e0303.py

```diff
@@ -0,0 +1,44 @@
+"""Check bad value returned by __len__ """
+
+# pylint: disable=too-few-public-methods, missing-docstring, no-self-use
+import sys
+
+import six
+
+
+class FirstGoodLen(object):
+    """__len__ returns <type 'int'>"""
+
+    def __len__(self):
+        return 0
+
+
+class SecondGoodLen(object):
+    """__len__ returns <type 'long'>"""
+
+    def __len__(self):
+        return sys.maxsize + 1
+
+
+class LenMetaclass(type):
+    def __len__(cls):
+        return 1
+
+
+@six.add_metaclass(LenMetaclass)
+class ThirdGoodLen(object):
+    """Length through the metaclass."""
+
+
+class FirstBadLen(object):
+    """ __len__ returns a negative integer """
+
+    def __len__(self):
+        return -1
+
+
+class SecondBadLength(object):
+    """ __len__ float """
+
+    def __len__(self):
+        return 3.0
```

Comment:
Can you add a test case with something that can't be inferred? What I'm doing is I try to do an import from a module that doesn't exist and use a member from that module in places where I want to test that we're not emitting anything when we can't infer things.

Something like:

   from missing import Missing

   ``__len__ = lambda self: Missing``